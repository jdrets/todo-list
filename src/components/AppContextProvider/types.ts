import React from 'react'

export interface TasksTypes {
  status: string
  priority: string
  dueDate?: string
  title: string
  description: string
}

export interface AppContextProviderTypes {
  tasks: TasksTypes[]
  setTasks: React.SetStateAction<any>
  newTaskModal: boolean
  setNewTaskModal: React.SetStateAction<any>
}
