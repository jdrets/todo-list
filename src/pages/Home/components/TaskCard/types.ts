export interface TaskTypes {
  status: string,
  priority: string,
  title: string,
  description: string
  dueDate?: string
}
