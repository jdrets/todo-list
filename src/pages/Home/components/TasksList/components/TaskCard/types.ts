export interface TaskTypes {
  id?: number,
  status: string,
  priority: string,
  title: string,
  description: string
  dueDate?: string
}

export interface TaskCardTypes {
  task: TaskTypes,
  setSelectedTask: (task: TaskTypes) => void
}
