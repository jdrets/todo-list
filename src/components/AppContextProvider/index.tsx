import React, { FunctionComponent, useState, createContext, useEffect } from 'react'
import _ from 'lodash'

import TaskService from '../../services/taskService'
import { AppContextProviderTypes } from './types'

export const PageContext = createContext({} as AppContextProviderTypes)

const AppContextProvider: FunctionComponent<any> = ({ children, value = {} }) => {
  const [tasks, setTasks] = useState([])
  const [showCreateTaskModal, setShowCreateTaskModal] = useState(false)

  useEffect(() => {
    const storage = TaskService.get()
    setTasks(storage)
  }, [])

  const updateContextWithNewTask = (task) => {
    const newTasks = _.clone(tasks)
    newTasks.push(task)
    setTasks(_.reverse(newTasks))
  }

  const contextValues = {
    tasks,
    showCreateTaskModal,
    updateContextWithNewTask,
    setShowCreateTaskModal,
    ...value
  }

  return (
    <PageContext.Provider value={contextValues}>{children}</PageContext.Provider>
  )
}

export default AppContextProvider
