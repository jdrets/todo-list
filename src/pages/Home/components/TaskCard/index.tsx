import React, { FunctionComponent } from 'react'

import {
  Wrapper,
  Title,
  Description
} from './styles'
import { TaskTypes } from './types'
import List from './components/List'

const TaskCard: FunctionComponent<TaskTypes> = ({ title, description, status, priority, dueDate }) => {
  return (
    <Wrapper>
      <List status={status} priority={priority} dueDate={dueDate} />
      <Title>{title}</Title>
      <Description>{description}</Description>
    </Wrapper>
  )
}

export default TaskCard
