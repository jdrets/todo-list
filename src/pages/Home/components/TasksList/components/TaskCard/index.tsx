import React, { FunctionComponent } from 'react'

import {
  Wrapper,
  Title,
  Description
} from './styles'
import { TaskCardTypes } from './types'
import List from './components/List'

const TaskCard: FunctionComponent<TaskCardTypes> = ({
  task,
  setSelectedTask,
  isDragging
}) => {
  const handleClick = () => {
    setSelectedTask(task)
  }

  return (
    <Wrapper onClick={handleClick} isDragging={isDragging}>
      <List status={task.status} priority={task.priority} dueDate={task.dueDate} />
      <Title>{task.title}</Title>
      <Description>{task.description}</Description>
    </Wrapper>
  )
}

export default TaskCard
