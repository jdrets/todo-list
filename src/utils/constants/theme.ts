const palette = {
  purpleDark: '#4B00BC',
  purple: '#6600FF',
  purpleMedium: '#8533FF',
  purpleLight: '#E3D0FF',
  green: '#59BC6F',
  greenLight: '#E6F5E2',
  red: '#EB5757',
  redLight: '#FCE4E4',
  yellow: '#DD9900',
  yellowLight: '#FEF1D8',
  white: '#FFFFFF',
  black: '#000000',
  gray: '#898989',
  grayLight: '#E0E0E0'
}

const theme = {
  colors: {
    primaryColor: palette.purple,
    primaryColorMedium: palette.purpleMedium,
    primaryColorLight: palette.purpleLight,
    primaryColorDark: palette.purpleDark,
    ...palette
  },
  layout: {
    breakpoints: {
      medium: '768px'
    },
    mobile: {
      width: '100%',
      padding: '0 24px'
    },
    desktop: {
      width: '1050px'
    }
  }
}

export default theme
