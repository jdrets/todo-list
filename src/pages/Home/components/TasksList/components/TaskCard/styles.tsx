import styled from 'styled-components'

import theme from '../../../../../../utils/constants/theme'

export const Wrapper = styled.article`
  border-radius: 6px;
  border: 1px solid ${theme.colors.grayLight};
  cursor: pointer;
  padding: 16px;
  transition: border-color 0.3s;
  
  &:hover {
    border-color: ${theme.colors.primaryColorLight};
  }
  
  &:active {
    border-color: ${theme.colors.primaryColor};
  }
`

export const Title = styled.div`
  color: ${theme.colors.primaryColor};
  font-size: 18px;
  font-weight: 500;
  margin-top: 8px;
  
  @media(min-width: ${theme.layout.breakpoints.medium}) {
    font-size: 24px;
    margin-top: 12px;
  }
`

export const Description = styled.div`
  font-size: 12px;
  font-weight: 300;
  line-height: 18px;
  margin-top: 6px;

  @media(min-width: ${theme.layout.breakpoints.medium}) {
    font-size: 14px;
    line-height: 22px;
  }
`
