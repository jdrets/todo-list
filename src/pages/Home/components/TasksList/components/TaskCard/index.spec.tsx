import React from 'react'
import { ThemeProvider } from 'styled-components'

import { render } from '@testing-library/react'

import TaskCard from './index'
import theme from '../../../../../../utils/constants/theme'
import AppContextProvider from '../../../../../../components/AppContextProvider'

const setup = () => (
  render(
    <ThemeProvider theme={theme}>
      <AppContextProvider>
        <TaskCard
          status='dummy status'
          priority='dummy priority'
          title='dummy title'
          description='dummy description'
          dueDate='dummy due date'
        />
      </AppContextProvider>
    </ThemeProvider>
  )
)

describe('<TaskCard />', () => {
  test('should match with snapshot', () => {
    const { container } = setup()

    expect(container).toMatchSnapshot()
  })
})
