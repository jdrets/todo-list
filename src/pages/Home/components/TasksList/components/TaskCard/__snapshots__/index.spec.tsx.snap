// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskCard /> should match with snapshot 1`] = `
<div>
  <article
    class="sc-bdnylx jafSpd"
  >
    <div
      class="sc-eCApGN biFMTw"
    >
      <div
        class="sc-jSFkmK ikJtKn"
      >
        <div
          class="sc-gKAblj fWNjKB"
        >
          <span
            class="sc-hKFyIo jqsNzg"
            color="#000000"
            height="22px"
            rotate="0"
            width="22px"
          >
            <svg
              class=""
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <path
                d="M307.72 302.27a8 8 0 01-3.72-6.75V80a48 48 0 00-48-48h0a48 48 0 00-48 48v215.52a8 8 0 01-3.71 6.74 97.51 97.51 0 00-44.19 86.07A96 96 0 00352 384a97.49 97.49 0 00-44.28-81.73zM256 112v272"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-miterlimit="10"
                stroke-width="32"
              />
              <circle
                cx="256"
                cy="384"
                r="48"
              />
            </svg>
          </span>
        </div>
        <div
          class="sc-iCoHVE eZjSys"
        >
          Priority: 
          <span
            class="sc-fujyUd eqjiKa"
          >
            dummy priority
          </span>
        </div>
      </div>
      <div
        class="sc-jSFkmK ikJtKn"
      >
        <div
          class="sc-gKAblj fWNjKB"
        >
          <span
            class="sc-hKFyIo jqsNzg"
            color="#000000"
            height="22px"
            rotate="0"
            width="22px"
          >
            <svg
              class=""
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <path
                d="M394.23 197.56a300.43 300.43 0 00-53.37-90C301.2 61.65 249.05 32 208 32a16 16 0 00-15.48 20c13.87 53-14.88 97.07-45.31 143.72C122 234.36 96 274.27 96 320c0 88.22 71.78 160 160 160s160-71.78 160-160c0-43.3-7.32-84.49-21.77-122.44zm-105.9 221.13C278 429.69 265.05 432 256 432s-22-2.31-32.33-13.31S208 390.24 208 368c0-25.14 8.82-44.28 17.34-62.78 4.95-10.74 10-21.67 13-33.37a8 8 0 0112.49-4.51A126.48 126.48 0 01275 292c18.17 24 29 52.42 29 76 0 22.24-5.42 39.77-15.67 50.69z"
              />
            </svg>
          </span>
        </div>
        <div
          class="sc-iCoHVE eZjSys"
        >
          Status: 
          <span
            class="sc-pNWxx kbZvre"
          >
            dummy status
          </span>
        </div>
      </div>
      <div
        class="sc-jSFkmK ikJtKn"
      >
        <div
          class="sc-gKAblj fWNjKB"
        >
          <span
            class="sc-hKFyIo jqsNzg"
            color="#000000"
            height="22px"
            rotate="0"
            width="22px"
          >
            <svg
              class=""
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <rect
                fill="none"
                height="384"
                rx="48"
                stroke="currentColor"
                stroke-linejoin="round"
                stroke-width="32"
                width="416"
                x="48"
                y="80"
              />
              <circle
                cx="296"
                cy="232"
                r="24"
              />
              <circle
                cx="376"
                cy="232"
                r="24"
              />
              <circle
                cx="296"
                cy="312"
                r="24"
              />
              <circle
                cx="376"
                cy="312"
                r="24"
              />
              <circle
                cx="136"
                cy="312"
                r="24"
              />
              <circle
                cx="216"
                cy="312"
                r="24"
              />
              <circle
                cx="136"
                cy="392"
                r="24"
              />
              <circle
                cx="216"
                cy="392"
                r="24"
              />
              <circle
                cx="296"
                cy="392"
                r="24"
              />
              <path
                d="M128 48v32M384 48v32"
                fill="none"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="32"
              />
              <path
                d="M464 160H48"
                fill="none"
                stroke="currentColor"
                stroke-linejoin="round"
                stroke-width="32"
              />
            </svg>
          </span>
        </div>
        <div
          class="sc-iCoHVE eZjSys"
          data-testid="card-due-date"
        >
          Due date: 
          <span
            class="sc-jrsJCI laYCos"
          >
            dummy due date
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-gtssRu hJpPPJ"
    >
      dummy title
    </div>
    <div
      class="sc-dlnjPT lmTqiz"
    >
      dummy description
    </div>
  </article>
</div>
`;
